### ISSUE TITLE
runlog

### ISSUE THREAD

**Issue by MarkTrinder** [naturalengland/Seabird_PVA_Tool#8]
The run log is not working - I get a garbled html document but not the word one. I have tried closing the browser and re-running but still no joy.

**Comment by dcmo** [comment]
Is the log file generated if you run the tool online? 
http://ec2-54-229-75-12.eu-west-1.compute.amazonaws.com/shiny/seabirds/PVATool/R/

Not all browsers behave in the same way, have you tried it using a different browser? Chrome is recommended.

**Comment by MarkTrinder** [comment]
Sorry – should have been more explicit. This was running it online and using Chrome. The log file was produced for the first run but thereafter failed. And closing Chrome and re-starting didn’t seem to fix it (a cookie hangover effect?).

**Comment by dcmo** [comment]
I am unable to replicate this (online or locally). I have tried running the tool online several times using chrome and the log file is generated normally each time. I have also tried using Edge and that also behaves as expected. Clearing your browser cache may be the next step to try.

**Comment by MarkTrinder** [comment]
Ok – will try that
thanks

**Comment by MarkTrinder** [comment]
still having problems - It seems that the first time I do a run it works fine, but if I then try to change a setting and re-run the log fail (server error). Or even just re-running without making any changes the same thing happens.

**Comment by MarkTrinder** [comment]
actually - even if I close Chrome and reset parameters this still happens.

**Comment by dcmo** [comment]
I tried several runs this morning with 50 simulations - all working as expected and log files generated normally - but when I tried running with 5000 simulations, I got the same server error. I have no access to the server so can't address any server errors. The alternative would be to try running it locally in RStudio.

**Comment by MarkTrinder** [comment]
Will that generate a log file? As I will be using this model for wind farm case work I need to be able to provide the SNCB with a log file so they can validate the results.


Mark Trinder
Principal Project Ornithologist
0141 342 5404  | 
07824 448 031 
93 South Woodside Road, Glasgow, G20 6NT  
www.macarthurgreen.com
​We help combat the climate crisis by operating a carbon negative business. 

​IMPORTANT NOTICE: This notice applies to this email and to any other email subsequently sent by anyone at MacArthur Green Ltd and appearing in the same chain of email correspondence. This e-mail and its attachments (if any) are confidential, protected by copyright and may be privileged. If you receive this e-mail in error, notify us immediately by reply e-mail, delete it and do not use, disclose or copy it (as it may be unlawful for you to do so). Unless we expressly say otherwise in this e-mail, this e-mail does not create, form part of, or vary, any contractual or unilateral obligation.   No liability is accepted for viruses and it is your responsibility to scan attachments (if any). Where this e-mail is unrelated to the business of MacArthur Green Ltd, the opinions expressed within this e-mail are the opinions of the sender and do not necessarily constitute those of MacArthur Green Ltd.   Company number SC354076. Registered office 93 South Woodside Road, Glasgow G20 6NT
​
 PLEASE CONSIDER THE ENVIRONMENT BEFORE PRINTING THIS E-MAIL.
​

**Comment by MarkTrinder** [comment]
Will that generate a log file? As I will be using this model for wind farm case work I need to be able to provide the SNCB with a log file so they can validate the results.

**Comment by dcmo** [comment]
Yes, it will, you just need to ensure the folder structure is the same (everything under the R folder) and the right packages are installed. Open app.R and click the 'run app' button in RStudio (make sure it is set to run in a browser not the viewer window).

**Comment by MarkTrinder** [comment]
ok - will give that a shot, thanks

**Comment by MarkTrinder** [comment]
so - running locally, using chrome, it initially still failed to create a log file. this was the console error:

> runApp('2 POPULATION MODELS/CEH PVA for NE/Seabird_PVA_Tool-master/Seabird_PVA_Tool-master/R')

Listening on http://127.0.0.1:7838
Quitting from lines 18-20 (NEPVAlog.Rmd) 
Warning: Error in [: subscript out of bounds

eventually tracked this down to the packages being stored in different locations and therefore not being accessed when required. I have had related problems with admin rights and where packages are installed, which may be a quirk of my IT setup or may be more common. perhaps something to consider for the future?

thanks for you help

**Comment by dcmo** [comment]
Thanks - I can see that's the line in the markdown file that lists the package versions so if any were not found it would complain. I'll make a note to see if there's a better way to check packages before the markdown section executes.

**Comment by bogsnork** [comment]
Hi, sorry only just noticed this.  Thanks for raising the issue @MarkTrinder and thanks for looking into it @dcmo .  

I've just had the same issue.  I'll have a look at the server and see if there's an issue there.
